android/envoy/build.gradle

```diff
@@ -64,78 +68,80 @@ exec {
 version = stdout.toString().trim()
 group = 'org.greatfire'
 
-task sourcesZip {
-  def src = 'https://github.com/greatfire/envoy/archive/refs/tags/' + project.version + '.zip'
-  def destdir = project.getBuildDir().toString() + "/outputs/aar"
-  def destfile = "$destdir/envoy-" + project.version + "-sources.zip"
-  doLast {
-    def url = new URL(src)
-    def f = new File(destfile)
-    if (f.exists()) {
-      println "file $destfile already exists, skipping download"
-    } else {
-      mkdir "$destdir"
-      println "Downloading $destfile from $url..."
-      url.withInputStream { i -> f.withOutputStream { it << i } }
+tasks.register('sourcesZip') {
+    def src = 'https://github.com/greatfire/envoy/archive/refs/tags/' + project.version + '.zip'
+    def destdir = project.getBuildDir().toString() + "/outputs/aar"
+    def destfile = "$destdir/envoy-" + project.version + "-sources.zip"
+    doLast {
+        def url = new URL(src)
+        def f = new File(destfile)
+        if (f.exists()) {
+            println "file $destfile already exists, skipping download"
+        } else {
+            mkdir "$destdir"
+            println "Downloading $destfile from $url..."
+            url.withInputStream { i -> f.withOutputStream { it << i } }
+        }
     }
-  }
 }
 
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
 
-task writeNewPom {
-    pom {
-        project {
-            name 'Envoy'
-            packaging 'aar'
-            artifactId 'envoy'
-            inceptionYear '2018'
-            url 'https://github.com/greatfire/envoy'
-            description 'C and Java Library derived from Chromium cronet'
-            licenses {
-                license {
-                    name 'The Apache Software License, Version 2.0'
-                    url 'https://github.com/greatfire/envoy/blob/master/LICENSE'
-                    distribution 'repo'
-                }
-                license {
-                    name 'BSD-3-clause'
-                    url 'https://chromium.googlesource.com/chromium/src/+/refs/heads/main/LICENSE'
-                    distribution 'repo'
-                }
-            }
-            developers {
-                developer {
-                    id = 'greatfire'
-                    name = 'GreatFire'
-                    email = 'support@greatfire.org'
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
```

Comment:
I assume this will publish to maven automatically rather than generating a .pom file for us to publish it manually?

Reply:
TBH, I couldn't test the `maven-publish` integration. (How should I?)
It probably needs more work.

But it is a start, which at least lets Gradle compile.

The `maven` plugin isn't supported anymore, so we gotta move.